src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.specialized.BlobLeaseAsyncClient;
+import com.azure.storage.blob.specialized.BlobLeaseClientBuilder;
+import org.slf4j.Logger;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class ContainerProcessor {
+
+    private static final Logger LOGGER = getLogger(ContainerProcessor.class);
+
+    private final BlobProcessor blobProcessor;
+
+    public ContainerProcessor(BlobProcessor blobProcessor) {
+        this.blobProcessor = blobProcessor;
+    }
+
+    public void process(BlobContainerAsyncClient containerClient) {
+        String containerName = containerClient.getBlobContainerName();
+        AtomicInteger atomInt = new AtomicInteger(0);
+
+        LOGGER.info("Processing container {}", containerName);
+
+        containerClient
+            .listBlobs()
+            .subscribe(
+                blob -> {
+                    processBlobs(containerClient, blob);
+                    atomInt.updateAndGet(operand -> ++operand);
+                },
+                null, // error consumer. can be implemented later
```

Comment:
We don't need to bring the whole job in at once, but when we're adding a class that's responsible for some task, that class needs to do its thing properly. Error handling is quite a crucial element of any processing and it can't be added later, as if it was additional functionality.